src/Corax/IndexWriter.cs

```diff
@@ -502,31 +510,71 @@ public long Index(Slice id, Span<byte> data)
 
         private IndexedField GetDynamicIndexedField(ByteStringContext context, ref IndexEntryReader.DynamicFieldEnumerator it)
         {
-            _dynamicFieldsTerms ??= new();
+            _dynamicFieldsTerms ??= new(SliceComparer.Instance);
             using var _ = Slice.From(context, it.CurrentFieldName, out var slice);
 
             if (_dynamicFieldsTerms.TryGetValue(slice, out var indexedField))
                 return indexedField;
-            
+
             
             var clonedFieldName = slice.Clone(context);
-
-            if (_dynamicFieldsMapping != null && _dynamicFieldsMapping.TryGetByFieldName(slice, out var binding))
+            
+            if (_dynamicFieldsMapping is null || _persistedDynamicFieldsAnalyzers is null)
             {
+                CreateDynamicField(null, FieldIndexingMode.Normal);
+                return indexedField;
+            }
+            
+            var persistedAnalyzer = _persistedDynamicFieldsAnalyzers.Read(slice);
+            if (_dynamicFieldsMapping?.TryGetByFieldName(slice, out var binding) is true)
+            {
+                
+                indexedField = new IndexedField(Constants.IndexWriter.DynamicField, binding.FieldName, binding.FieldNameLong, binding.FieldNameDouble, binding.Analyzer, binding.FieldIndexingMode, binding.HasSuggestions);
+
+                if (persistedAnalyzer != null && binding.FieldIndexingMode != (FieldIndexingMode)persistedAnalyzer.Reader.ReadByte())
+                {
+                    throw new InvalidDataException("Your index is dynamically changing analyzer in dynamic field. We do not support it.");
```

Comment:
in dynamic field -> of dynamic field. Maybe include the field name, and the difference in the usage options as well?

Reply:
I wonder if throwing something like: `Inconsistent dynamic field creation options were detected. Field 'Foo' was created with 'x' analyzer but now 'y' analyzer was specified. This is not supported`  wouldn't make it more clear to a user

Reply:
done